apps/test/unit/templates/sectionProgress/progressTables/ProgressTableLessonNumberTest.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import {expect} from '../../../../util/reconfiguredChai';
+import {shallow} from 'enzyme';
+import ProgressTableLessonNumber from '@cdo/apps/templates/sectionProgress/progressTables/ProgressTableLessonNumber';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import ReactTooltip from 'react-tooltip';
+import sinon from 'sinon';
+import color from '@cdo/apps/util/color';
+
+const DEFAULT_PROPS = {
+  name: 'Intro to Problem Solving',
+  number: 1,
+  lockable: false,
+  highlighted: false,
+  onClick: () => {},
+  includeArrow: false,
+  isAssessment: false
+};
+
+const setUp = (overrideProps = {}) => {
+  const props = {...DEFAULT_PROPS, ...overrideProps};
+  return shallow(<ProgressTableLessonNumber {...props} />);
+};
+
+describe('ProgressTableLessonNumber', () => {
+  it('renders tooltip with an icon if isAssessment is true', () => {
+    const wrapper = setUp({isAssessment: true});
+    const tooltipIcon = wrapper.find({icon: 'check-circle'});
+    expect(tooltipIcon).to.have.length(1);
+  });
+
+  it('renders tooltip without an icon if isAssessment is false', () => {
+    const wrapper = setUp({isAssessment: false});
+    const tooltipIcon = wrapper.find({icon: 'check-circle'});
+    expect(tooltipIcon).to.have.length(0);
+  });
+
+  it('renders the name in the tooltip', () => {
+    const wrapper = setUp();
+    const tooltipComponent = wrapper.find(ReactTooltip);
+    expect(tooltipComponent.contains(DEFAULT_PROPS.name)).to.equal(true);
+  });
+
+  it('calls onClick when clicked', () => {
+    const onClickSpy = sinon.spy();
+    const wrapper = setUp({onClick: onClickSpy});
+    wrapper.simulate('click');
+    expect(onClickSpy).to.have.been.called;
+  });
+
+  it('displays highlighed styles when highlighted is true', () => {
+    const wrapper = setUp({highlighted: true});
+    expect(wrapper.props().style.backgroundColor).to.equal(color.teal);
+  });
+
+  it('displays line and arrow when includeArrow is true', () => {
```

Comment:
I spent a while in analysis-paralysis about the best way to test that this arrow is present on the page. I don't love this, open to other options.

Reply:
ooh, this is a great opportunity to try the method i was proposing when we chatted of factoring out subcomponents for enzyme to find. i went ahead and took a stab at it to makes sure it works and so you can see what i mean. imo this is very clean for testing and even improves the component code.

if you like this approach, could you try to figure out how to only export the `LessonArrow` when `IN_UNIT_TEST == true`? if you search the code for that string you can see some different examples of using it, but i spent a few minutes trying to get it to work and couldn't so i'm punting to you. ;-)

Reply:
Cool - yeah this was an approach that I started, I factored the arrow out into a separate component but then realized I'd have to export it in order to use it in the test and wasn't sure if that was a good enough reason to export another thing. But it is clean. I'll take a stab at finishing this up. Another thing that Maddie just informed me of is that for eyes (and some unit tests) we use `className='ui-test-<description>'` as a way to more easily select elements in tests. So this is a pattern we could also choose to follow in the future.

Reply:
cool. i'm optimistic that you'll be able to figure out how to export the arrow component only in unit tests, but if not, i personally would prefer exporting subcomponents for testing over using special class names in the code, even if we can't keep them "private". my thinking is that if we need to select it, it must represent a useful encapsulation of functionality, and therefore deserves to be its own "thing". 

also, aside from making the code cleaner, it makes it more reusable. if someone were to say "i need an arrow like that one in the progress tables for my component", they would look in this component and easily find what they want. we have a ton of copy/pasted code in our codebase, because i think currently when people run into a situation like that, they just copy the block of code they want into their component. but if they see that it's already its own encapsulated component, they will be more inclined to just move it to its own file for reuse in their component.

react strongly favors composition over inheritance for code reuse, and i think the more we break our components down, the easier it will be for other devs to see how to re-compose them into new components.

we may find that there are a lot of situations where using class names just makes more sense. but i would like for us to start with the assumption that there will never be such a case for the purposes of this project and see how it goes. basically, i think this question of "how do i select a thing without using class names" might actually force us to improve our component architecture and design strategy. but i could be wrong. let's try it and see!